src/arfsdao.ts

```diff
@@ -571,4 +571,140 @@ export class ArFSDAO {
+	// 		}
+
+	// 		return tag.value;
+	// 	}
+
+	// 	edgeToEntity(edge: GQLEdgeInterface): ArFSEntity {
+	// 		const { tags } = edge.node;
+
+	// 		return {
+	// 			appName: this.getTagValue('App-Name', tags),
+	// 			appVersion: this.getTagValue('App-Version', tags),
+	// 			arFS: this.getTagValue('ArFS', tags),
+	// 			contentType: this.getTagValue('Content-Type', tags),
+	// 			driveId: this.getTagValue('Drive-Id', tags),
+	// 			entityType: this.getTagValue('Entity-Type', tags),
+	// 			name: '',
+	// 			txId: edge.node.id,
+	// 			unixTime: +this.getTagValue('Unix-Time', tags),
+	// 			syncStatus: 0
+	// 		};
+	// 	}
+
+	// 	async getAllEntitiesForDriveId(driveId: DriveID): Promise<ArFSEntity[]> {
+	// 		let hasNextPage = true;
+	// 		let cursor = '';
+	// 		let allEntities: ArFSEntity[] = [];
+
+	// 		while (hasNextPage) {
+	// 			const gqlQuery = buildQuery([{ name: 'Drive-Id', value: driveId }], cursor);
+	// 			const response = await this.arweave.api.post(graphQLURL, gqlQuery);
+	// 			const { data } = response.data;
+	// 			const { transactions } = data;
+	// 			const edges: GQLEdgeInterface[] = transactions.edges;
+	// 			hasNextPage = transactions.pageInfo.hasNextPage;
+
+	// 			for (const edge of edges) {
+	// 				cursor = edge.cursor;
+
+	// 				const baseEntity = this.edgeToEntity(edge);
+
+	// 				allEntities = [...allEntities, baseEntity];
+
+	// 				edge.node.tags;
+	// 				// switch (baseEntity.entityType) {
+	// 				// 	case 'file':
+
+	// 				// 		break;
+
+	// 				// 	default:
+	// 				// 		break;
+	// 				// }
+	// 			}
+	// 		}
+
+	// 		return allEntities;
+	// 	}
+}
+
+export class ArFSPublicDrive extends ArFSEntity implements ArFSDriveEntity {
+	constructor(
+		readonly appName: string,
+		readonly appVersion: string,
+		readonly arFS: string,
+		readonly contentType: string,
+		readonly driveId: string,
+		readonly entityType: string,
+		readonly name: string,
+		readonly txId: string,
+		readonly unixTime: number,
+		readonly drivePrivacy: string,
+		readonly rootFolderId: string
+	) {
+		super(appName, appVersion, arFS, contentType, driveId, entityType, name, 0, txId, unixTime);
+	}
+}
+
+export class ArFSPublicDriveBuilder {
+	appName?: string;
+	appVersion?: string;
+	arFS?: string;
+	contentType?: ContentType;
+	driveId?: DriveID;
+	entityType?: EntityType;
+	name?: string;
+	txId?: TransactionID;
+	unixTime?: number;
+	drivePrivacy?: DrivePrivacy;
+	rootFolderId?: FolderID;
+
+	build(): ArFSPublicDrive {
+		if (
+			this.appName &&
+			this.appName.length &&
+			this.appVersion &&
+			this.appVersion.length &&
+			this.arFS &&
+			this.arFS.length &&
+			this.contentType &&
+			this.contentType.length &&
+			this.driveId &&
+			this.driveId.length &&
+			this.entityType &&
+			this.entityType.length &&
+			this.name &&
+			this.name.length &&
+			this.txId &&
+			this.txId.length &&
+			this.unixTime &&
+			this.drivePrivacy &&
+			this.drivePrivacy.length &&
+			this.rootFolderId &&
+			this.rootFolderId.length
```

Comment:
We can simplify this a little like so
```suggestion
			this.appName?.length &&
			this.appVersion?.length &&
			this.arFS?.length &&
			this.contentType?.length &&
			this.driveId?.length &&
			this.entityType?.length &&
			this.name?.length &&
			this.txId?.length &&
			this.unixTime &&
			this.drivePrivacy?.length &&
			this.rootFolderId?.length
```